components/layout/flex.rs

```diff
@@ -262,18 +276,30 @@ impl FlexFlow {
         } else {
             self.block_flow.fragment.border_box.size.inline
         };
+
         for kid in &mut self.items {
-            let base = flow::mut_base(flow_ref::deref_mut(&mut kid.flow));
+            let grow_by = if grow_space_available < Au(0) && shrink_count > 0_f32 {
+                // TODO shrink based on scaled_shrink_factor
+                let percent_shrink = kid.flow.as_block().fragment.style.get_position().flex_shrink / shrink_count;
+                Au::to_f32_px(grow_space_available) * percent_shrink
+            }else if grow_space_available > Au(0) && grow_count > 0_f32 {
+                let percent_grow = kid.flow.as_block().fragment.style.get_position().flex_grow / grow_count;
+                Au::to_f32_px(grow_space_available) * percent_grow
+            }else {
+                0_f32
+            };
 
-            base.block_container_inline_size = even_content_inline_size;
+            let base = flow::mut_base(flow_ref::deref_mut(&mut kid.flow));
```

Comment:
Hmm, I'm not a big fan of pushing things down instead of grabbing from the parent, but the old code did this too, so this is OK.